core/packages/contracts/src/Registry.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+import {AccessControl} from "openzeppelin/access/AccessControl.sol";
+
+contract Registry is AccessControl {
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+    bytes32 public constant REGISTER_ROLE = keccak256("REGISTER_ROLE");
+
+    mapping(bytes32 contractID => address) registry;
+
+    constructor() {
+        _grantRole(ADMIN_ROLE, msg.sender);
+        _setRoleAdmin(ADMIN_ROLE, ADMIN_ROLE);
+        _setRoleAdmin(REGISTER_ROLE, ADMIN_ROLE);
```

Comment:
will `_grantRole(REGISTER_ROLE, msg.sender);` also required here?

Reply:
The deployer address does that in `DeployScript.sol`